### PR TITLE
Optional final production model fitting procedure after CV

### LINKED ISSUE #217: Final Model Training
**Is your feature request related to a problem? Please describe.**

After evaluating whatever models you evaluate with DrEvalPy, it would be great to also generate a model based on the optimal parameters and trained on all available data.

**Describe the solution you would like**

Add an optional parameter to train a model using all data (i.e. include validation set).  Make sure to have this as an optional parameter, as this will cause an immense additional space requirement.

**Additional context**

I don't think this needs more context ;)

### PR DESCRIPTION
**Pull request by PascalIversen**
closes #217
I decided for a proper retuning on the full dataset instead of any hacks. (will approx. take as long as one fold)
 
also sneaking in tissue mapping tests which I forgot to add.

TODO:
testing

### CODE REVIEW

File: drevalpy/models/SRMF/srmf.py [hunk 1]
```diff
@@ -294,3 +298,59 @@ def load_drug_features(self, data_path: str, dataset_name: str) -> FeatureDatase
         :returns: FeatureDataset containing the drug fingerprint features
         """
         return load_drug_fingerprint_features(data_path, dataset_name, fill_na=True)
+
+    def save_model(self, directory: str) -> None:
```
Comment (JudithBernett) [inline drevalpy/models/SRMF/srmf.py:302]: why is this save_model and note save?

File: drevalpy/models/SRMF/srmf.py [hunk 2]
```diff
@@ -294,3 +298,59 @@ def load_drug_features(self, data_path: str, dataset_name: str) -> FeatureDatase
...
+                    "lambda_d": self.lambda_d,
+                    "lambda_c": self.lambda_c,
+                    "max_iter": self.max_iter,
+                    "seed": self.seed,
+                    "training_mean": self.training_mean,
+                },
+                f,
+            )
+
+    def load_model(self, directory: str) -> None:
```
Comment (JudithBernett) [inline drevalpy/models/SRMF/srmf.py:332]: why does this return none and is load_model and not load?

File: drevalpy/models/drp_model.py [hunk 3]
```diff
@@ -170,6 +170,36 @@ def load_drug_features(self, data_path: str, dataset_name: str) -> FeatureDatase
         :returns: FeatureDataset or None
         """
 
+    def save(self, directory: str) -> None:
```
Comment (JudithBernett) [inline drevalpy/models/drp_model.py:173]: Where are the save and load methods for the sklearn models?

File: tests/models/test_baselines.py [hunk 4]
```diff
@@ -121,6 +121,29 @@ def test_baselines(
...
+            preds_after = loaded_model.predict(
+                drug_ids=val_dataset.drug_ids,
+                cell_line_ids=val_dataset.cell_line_ids,
+                drug_input=drug_input,
+                cell_line_input=cell_line_input,
+            )
+            assert isinstance(preds_after, np.ndarray)
+            assert preds_after.shape == preds_before.shape
+        except NotImplementedError:
+            print(f"{model_name} does not implement save/load")
```
Comment (JudithBernett) [inline tests/models/test_baselines.py:145]: I think all our baselines should support this
```suggestion
```

**Review by JudithBernett — CHANGES_REQUESTED** [review]
:)))
